### PR TITLE
Fix add a section

### LINKED ISSUE #125: Adding a new section broken
### Expected behaviour

When adding a new section, I should them be able to select that section and edit properties

### Actual behaviour

When adding a new section, editing the title causes another section to be added with that title

### Steps to reproduce the behaviour

- start a new survey
- progress to design
- add a section
- select it
- edit the title

### PR DESCRIPTION
**Pull request by hamishtaplin**
### What is the context of this PR?

Fixes #125 

- sured up code that passes id's around
- refactored layout out of page components and into route definition to
  to make testing easier
- added some tests

### How to review 

As per bug repro steps

### CODE REVIEW

**Comment by samiwel** [comment]
@hamishtaplin There's still some weird things happening with the sidebar for me on this branch. Happy to demo.

File: src/containers/DesignSurvey.js [hunk 1]
```diff
@@ -1,55 +1,49 @@
...
     }
-  };
+    return updateItem(name, value);
+  }
 };
 
-export default connect(mapStateToProps, mapDispatchToProps, null, {
-  pure: true
-})(DesignSurveyPage);
+export default connect(mapStateToProps, mapDispatchToProps, null)(
```
Comment (WickyNilliams) [inline src/containers/DesignSurvey.js]: Think the `null` arg can be removed now?

File: src/index.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ import createHistory from "history/createHashHistory";
 import { configureStore, configurePersistedStore } from "store/configureStore";
 import App from "containers/App";
 
-let useStorage = false;
+let useStorage = true;
```
Comment (WickyNilliams) [inline src/index.js]: Should this be put back to false?

File: src/pages/CreateSurvey.test.js [hunk 3]
```diff
@@ -7,7 +7,10 @@ import { mountWithRouter } from "tests/utils/mountWithRouter";
 
 describe("CreateSurveyPage", () => {
   it("should render", () => {
-    const Page = mountWithRouter(<CreateSurveyPage meta={defaultState} />);
+    const Page = (
```
Comment (WickyNilliams) [inline src/pages/CreateSurvey.test.js]: This should be calling `mount`?

File: src/pages/DesignSurvey.js [hunk 4]
```diff
@@ -40,46 +42,48 @@ const DesignSurveyPage = ({
...
-    </SidebarPageLayout>
+    </EditLayout>
   );
 };
 
+DesignSurveyPage.propTypes = {
+  selected: PropTypes.shape({
+    id: PropTypes.string.isRequired,
+    type: PropTypes.string.isRequired,
+    item: PropTypes.oneOfType(CustomPropTypes.surveyItems.map(PropTypes.shape))
```
Comment (WickyNilliams) [inline src/pages/DesignSurvey.js]: See comments below about proptypes

File: src/proptypes/answer.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+/* eslint-disable camelcase */
+
+import PropTypes from "prop-types";
+
+export const answer = {
```
Comment (WickyNilliams) [inline src/proptypes/answer.js]: Why not directly wrap in `shape` here?

```js
export const answer = PropTypes.shape({
  description: PropTypes.string,
  displayName: PropTypes.string,
  mandatory: PropTypes.bool.isRequired,
  title: PropTypes.string,
  type: PropTypes.string.isRequired,
  q_code: PropTypes.string.isRequired
});
```

File: src/proptypes/index.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+import question from "proptypes/question";
+import answer from "proptypes/answer";
+import survey from "proptypes/survey";
+
+export const PropTypes = {
+  section,
+  question,
+  answer,
+  survey,
+  surveyItems: [section, question, answer]
```
Comment (WickyNilliams) [inline src/proptypes/index.js]: If you've wrapped section/question/answer etc `surveyItems` could simply be 

```js
PropTypes.oneOfType([section, question, answer]);
```

File: src/proptypes/survey.js [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import PropTypes from "prop-types";
+
+export const survey = {
+  items: {
```
Comment (WickyNilliams) [inline src/proptypes/survey.js]: If I remember rightly, this should be:

```js
{
  items : PropTypes.shape({
    // ...
  })
}
```

File: src/proptypes/survey.js [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+    sections: PropTypes.object,
```
Comment (WickyNilliams) [inline src/proptypes/survey.js]: You could reuse your section/question/answer proptypes here?

File: .eslintrc [hunk 9]
```diff
@@ -16,6 +16,7 @@
     "no-bitwise": 2,
     "no-caller": 2,
     "no-shadow": 0,
+    "no-console": 1,
```
Comment (hamishtaplin) [inline .eslintrc]: Yes, I use console all the time and don't want to add the extra comment as it slows me down.

I think this is fine as a warning assuming travis fails on warnings.

File: src/proptypes/survey.js (hunk 8 above)
Comment (hamishtaplin) [inline src/proptypes/survey.js]: Those proptypes are for a single `section` whereas this is a collection of sections (keyed by `id`). I probably can leverage those proptypes here, will look into it.

File: src/index.js (hunk 2 above)
Comment (hamishtaplin) [inline src/index.js]: I don't know

**Pull request merged**